### PR TITLE
Enhance mugc Functionality

### PR DESCRIPTION
**Pull request by bdwyertech**
- Allow targeting policies within a file (inverse of default)
- Add regex for flexible targeting
- Remove "-" prefix/policyname delimiter -- function name ends with policy name

### CODE REVIEW

File: tools/ops/mugc.py [hunk 1]
```diff
@@ -145,6 +155,9 @@ def setup_parser():
     parser.add_argument(
         '-c', '--config', dest="config_files", nargs="*", action='append',
         help="Policy configuration files(s)", default=[])
+    parser.add_argument(
+        "--targeted", action="store_true", default=False,
```
Comment (kapilt) [inline tools/ops/mugc.py]: not in love with the name, but drawing blanks on the moment on something better. the default is effectively --absent so this is perhaps --present ?

File: tools/ops/mugc.py [hunk 2]
```diff
@@ -155,6 +168,9 @@ def setup_parser():
     parser.add_argument(
         "--prefix", default="custodian-",
         help="The Lambda name prefix to use for clean-up")
+    parser.add_argument(
+        "--policy-regex", default="custodian-.*",
```
Comment (kapilt) [inline tools/ops/mugc.py]: this probably needs a leading ^ so it only matches from the beginning of the string.

```python
>>> import re
>>> re.search("custodian-.*", 'not-custodian-x')
<re.Match object; span=(4, 15), match='custodian-x'>
```

**Comment by kapilt** [comment]
thanks for the pr, I dig the features overall re more flexible selection and inverting the default (its been a common enough question in chat).

Fwiw,  Its probably worth noting there's a pending refactor for mugc, to support different providers, as well as additional execution modes (security hub actions, etc), we started tagging our lambda functions with mode and custodian version (0.8.44.3 afaicr), so some of the introspection we're doing on lambda permissions can go away. the intent on the refactor is to help us enable it for the main cli.

File: tools/ops/mugc.py (hunk 1 above)
Comment (bdwyertech) [reply tools/ops/mugc.py]: Agreed.  I was at a loss for naming as well

File: tools/ops/mugc.py (hunk 2 above)
Comment (bdwyertech) [reply tools/ops/mugc.py]: Yep you're right about that

**Comment by bdwyertech** [comment]
Glad to hear there's some refactoring going on for mugc, keeping it simple certainly helps though!

I've got this vendored in my own wrapper utility, along with quite a few other things... Looking forward to contributing what we can back, will be a very large deployment.

**Comment by kapilt** [comment]
overall lgtm, it would be nice to seem unit tests here, they fit well into extant.  re future plans here, they'll be in backwards compatible manner, and separate patjs, so not per se relevant.

**Comment by kapilt** [comment]
we probably need compatibility to the existing prefix syntax, to be honest regex is probably overkill here vs glob. and mugc really needs unit tests, given its used in production environments.

**Pull request merged**